### DISCUSSION TITLE
Git push completes before kudusync is done

### DISCUSSION THREAD

**Discussion by LennardF1989** [umbraco/Umbraco.Cloud.Issues#800] [Ideas]
### Issue description

It appears, if you do a large commit/push to an environment repository, the git-hook will sometimes return as completed (all A-OK signal), while in reality kudusync is still working on deploying the files to `C:\home\site\wwwroot`.

Is kudusync supposed to be handled asynchronously like this? I feel the push-hook should only complete if either something went wrong or when it's completely done.

I've look at the webhooks in Kudu, but haven't tried them yet - but I need a way to reliably tell when a deployment is done so other processes can start running.

**Comment by mclausen** [comment]
Hi @LennardF1989,

You’re absolutely right! The git-hook is fired when the git part of the operation is done, while the kudu-deploy can still be ongoing. Unfortunately, this is a built-in feature of Azure App Service git deployments. The [existing webhook](https://docs.umbraco.com/umbraco-cloud/deployments/deployment-webhook) functionality is a bit outdated and relies on the git-hook mechanism you mentioned.

The good news is that internally, we have mechanisms to hook into and publish better notifications. This means there’s potential for us to improve our implementation of this in the future!

I've added the feature to our internal trackers, and we'll provide update as soon as we have any 🎩 

Kind regards
@mclausen
